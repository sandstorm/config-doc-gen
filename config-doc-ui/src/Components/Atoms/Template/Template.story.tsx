import { storiesOf } from '@storybook/react';
import * as React from 'react';
import Template from '.';

storiesOf('Template', module)
  .add('default', () => (
    <Template/>
  ))
  .add('with template prop', () => (
    <Template template="Dude"/>
  ));
