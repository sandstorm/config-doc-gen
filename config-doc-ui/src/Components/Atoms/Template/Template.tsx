import * as React from 'react';
import {PickDefaultProps} from '../../../../types/defaultProps';

//
// Props
//
interface TemplateProps {
  readonly template: string;
}

type DefaultProps = PickDefaultProps<TemplateProps, 'template'>;

const defaultProps: DefaultProps = {
  template: 'Bro',
};

//
// State
//
interface TemplateState {
  readonly isCool: boolean;
}

const initialTemplateState: TemplateState = {
  isCool: true,
};

//
// Class
//
export default class Template extends React.PureComponent<TemplateProps, TemplateState> {
  public static readonly defaultProps = defaultProps;

  public constructor(props: TemplateProps) {
    super(props);
    this.state = initialTemplateState;
  }

  public render(): JSX.Element {
    return (
      <div className="template">
        <p>{this.getMessage()}</p>
        <p>{Template.getPureMessage(this.state.isCool, this.props.template)}</p>
        <p>{getPureMessageFromTheOutside(this.state.isCool, this.props.template)}</p>
      </div>
    );
  }

  private readonly getMessage = (): string => `Hey, ${this.state.isCool ? 'cool ' : ''}${this.props.template}!`;

  private static getPureMessage(isCool: boolean, template: string): string {
    return `Hey, ${isCool ? 'cool ' : ''}${template}!`;
  }
}

//
// Pure utility functions.
//

// It's almost always better to put them somewhere more 'global' (like 'src/Domain/')  where they can be shared and tested.
// Only put them here if they are really reeaally component specific.

function getPureMessageFromTheOutside(isCool: boolean, template: string): string {
  return `Hey, ${isCool ? 'cool ' : ''}${template}!`;
}
