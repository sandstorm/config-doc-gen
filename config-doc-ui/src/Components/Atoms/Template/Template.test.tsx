import * as React from 'react';
import {render} from 'react-dom';
import * as ReactTestRenderer from 'react-test-renderer';
import Template from '.';

describe('Template tests', () => {
  it('should render without crashing', () => {
    render(<Template/>, document.createElement('div'));
  });

  it('should match snapshot', () => {
    expect(ReactTestRenderer.create(<Template/>)).toMatchSnapshot();
  });

  it('should match snapshot with template props', () => {
    expect(ReactTestRenderer.create(<Template template="Dude"/>)).toMatchSnapshot();
  });
});
