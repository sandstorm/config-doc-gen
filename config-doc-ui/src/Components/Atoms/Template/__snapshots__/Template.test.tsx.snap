// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template tests should match snapshot 1`] = `
<div
  className="template"
>
  <p>
    Hey, cool Bro!
  </p>
  <p>
    Hey, cool Bro!
  </p>
  <p>
    Hey, cool Bro!
  </p>
</div>
`;

exports[`Template tests should match snapshot with template props 1`] = `
<div
  className="template"
>
  <p>
    Hey, cool Dude!
  </p>
  <p>
    Hey, cool Dude!
  </p>
  <p>
    Hey, cool Dude!
  </p>
</div>
`;
